deploy_and_run_integration_tests.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+set -euo pipefail
+
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script builds a Docker image from the current state of the local code,
+# pushes the image to a test registry (project 'gcp-variant-transforms-test')
+# and runs integration tests against that image.
+#
+# To run this test successfully:
+# - The user's gcloud credentials should be set; follow the steps at:
+#   https://cloud.google.com/genomics/install-genomics-tools
+#
+# - The user should have access to the 'gcp-variant-transforms-test' project.
+
+PROJECT="gcp-variant-transforms-test"
+GS_DIR="integration_test_runs"
+GREEN="\e[32m"
+RED="\e[31m"
+TEST_ARGUMENTS=""
+KEEP_IMAGE_OPT="--keep_image"
+IMAGE_TAG_OPT="--image_tag"
+SKIP_BUILD_OPT="--skip_build"
+image_tag=""
+keep_image=""
+skip_build=""
+
+#################################################
+# Prints a given message with a color.
+# Arguments:
+#   $1: The message
+#   $2: The text for the color, e.g., "\e[32m" for green.
+#################################################
+color_print() {
+  echo -n -e "$2"  # Sets the color to the given color.
+  echo "$1"
+  echo -n -e "\e[0m"  # Resets the color to no color.
+}
+
+#################################################
+# Prints the usage.
+#################################################
+usage() {
+  echo "Usage: $0 [${KEEP_IMAGE_OPT}] [${IMAGE_TAG_OPT} image_tag] "
+  echo "    [${SKIP_BUILD_OPT}] [... test script options ...]"
+  echo "  With ${KEEP_IMAGE_OPT} the tested image is not deleted at the end."
+  echo "  If ${IMAGE_TAG_OPT} is not set, a new image with the current time"
+  echo "    tag is created and used in tests, otherwise the given tag is used."
+  echo "  ${SKIP_BUILD_OPT} skips the build step so it has to be used with "
```

Comment:
Can we drop this flag and instead build iff 'image_tag' is empty? In other words, what's the use case for having non_empty 'image_tag' and skip_build=False in the integration test.

Reply:
I agree with you that in the integration test scenarios (e.g., before sending a PR) the --image_tag is enough but I can imagine other scenarios that this is useful. For example, when we want to make a release candidate, we may want to use this script, tag the image as RC (rather than with a test time tag) _but also_ build it.